src/Server.cpp

```diff
@@ -852,33 +950,97 @@ void handleCompleteRequestHeader(Server& server, int clientFd, Connection& conne
 	}
 
 	server.findTargetResource(connection, connection.m_request);
+	if (isCGIRequested(connection)) {
+		connection.m_request.hasCGI = true;
+		ProcessOps processOps;
+		CGIHandler cgiHandler(connection, processOps);
+		if (connection.m_request.httpStatus == StatusInternalServerError) {
+			connection.m_status = Connection::BuildResponse;
+			server.modifyEvent(clientFd, EPOLLOUT);
+			return;
+		}
+		cgiHandler.execute(connection.m_request, connection.location, processOps, server.getEpollWrapper().getEpollFd(),
```

Comment:
Since you pass the whole Connection in the constructor anyway you can just save everything you need as refrence (private member variable).

Then you would only need to pass the epollfd and the connections map
- request > saved
- location > saved
- processOps > passed in constructor

Reply:
Good point, done